lisa/util/process.py

```diff
@@ -308,6 +295,37 @@ def _filter_sudo_result(self, raw_input: str) -> str:
             self._log.debug(f'found error message in sudo: "{lines[0]}"')
         return raw_input
 
+    def _handle_windows_result(
+        self, process_result: ExecutionResult
+    ) -> ExecutionResult:
+        # remove extra controls in remote Windows
+        process_result.output = filter_ansi_escape(process_result.output)
+        process_result.stderr_output = filter_ansi_escape(process_result.stderr_output)
+
+        # Remove the extra line like below, Not sure where it's from,
+        # may need investigate more.
+        # 0;C:\Windows\system32\conhost.exe
+        if process_result.output:
+            process_result.output = process_result.output.replace(
+                "0;Administrator: C:\Windows\system32\conhost.exe", ""  # noqa: W605
+            )
+
+            # Unknown \x07 unicode character is returned by PowerShell
+            process_result.output = process_result.output.replace("\x07", "")
+
+        # Windows contains error in stdout, so we need to
+        # appropriately handle it.
+        if (
```

Comment:
I feel those changes are not reliable, it shouldn't be put in the lib level. If anything doesn't follow this pattern it's hard to fix. For example, some log show with Error, but it's considered as correct.
Let's move the check to each command. You can add some assert_output in ExecutableResult for convenient.

Reply:
I will push this change and reboot related change in another commit.

Reply:
The reboot can be improved next time, but this logic should be moved out from this change. The intermediate can be "not supported" but shouldn't be "incorrect".

Reply:
Figured out a way to make reboot robust. Will be the changes with NAT testcase

Reply:
I have a PR, with this fix, the process can return exit code now. The key thing is to disable tty for Windows.
https://github.com/microsoft/lisa/pull/1873